std/security/signer/validate_data.go

```diff
@@ -0,0 +1,46 @@
+package signer
+
+import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/x509"
+
+	enc "github.com/named-data/ndnd/std/encoding"
+	"github.com/named-data/ndnd/std/ndn"
+)
+
+// ValidateData verifies the signature of a Data packet with a certificate.
+func ValidateData(data ndn.Data, sigCovered enc.Wire, cert ndn.Data) (bool, error) {
```

Comment:
How is the client supposed to obtain the certificate?

Reply:
This is the low-level (known key) validation API. This is called by `TrustConfig`, which fetches the cert from network or local storage (if available).